src/App.tsx

```diff
@@ -0,0 +1,128 @@
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const filteredData = useFilter(countries, search);
+
+  console.info('Filtered countries:', filteredData);
+
+  return (
+    <div className="min-h-screen bg-gradient-to-b from-slate-50 to-white text-slate-900">
+      <header className="sticky top-0 z-40 border-b bg-white/70 backdrop-blur supports-[backdrop-filter]:bg-white/60">
+        <div className="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8 py-3 flex items-center justify-between">
+          <div className="flex items-center gap-3">
+            <div className="h-9 w-9 rounded-2xl bg-slate-900 text-white grid place-items-center shadow-sm">
+              CO₂
+            </div>
+            <h1 className="text-lg sm:text-xl font-semibold tracking-tight">
+              CO₂ Explorer
+            </h1>
+          </div>
+        </div>
+      </header>
+
+      <main className="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8 py-6 grid grid-cols-1 lg:grid-cols-12 gap-6">
+        <section className="lg:col-span-4 shadow-sm rounded-2xl">
+          <div className="pb-3 flex flex-col gap-2">
+            <h2 className="text-base">Countries</h2>
+            <Search value={search} onSearch={setSearch} />
+          </div>
+          <div className="pt-0">
+            <hr className="mb-3" />
+            <div className="h-[60vh] pr-3">
+              {isLoading ? (
+                <p>Loading...</p>
+              ) : (
+                <ul className="space-y-2">
+                  {filteredData.map(({ iso, data: { iso_code, data } }) => (
+                    <li key={iso}>
+                      <div
+                        className="w-full text-left rounded-xl px-3 py-2 border transition shadow-sm"
+                        onClick={() => setIsExpanding(!isExpanding)}
+                      >
+                        <div className="flex items-center gap-2">
+                          {isExpanding ? (
+                            <ChevronDown className="h-4 w-4 text-slate-400" />
+                          ) : (
+                            <ChevronRight className="h-4 w-4 text-slate-400" />
+                          )}
+
+                          <span className="font-medium">{iso}</span>
+                          <span className="text-xs text-slate-500">
+                            {iso_code ? `(${iso_code})` : '(N/A)'}
+                          </span>
+                          <span className="ml-auto text-xs text-slate-600">
+                            Pop: {data[data.length - 1].population}
+                          </span>
+                        </div>
+                        {isExpanding && (
+                          <div className="lg:col-span-8 shadow-sm">
+                            <h3 className="pb-3 text-base">Yearly data</h3>
+                            <Suspense fallback={<p>Loading...</p>}>
+                              <div className="rounded-xl border border-slate-200 overflow-hidden">
```

Comment:
Since your getData is a promise wrapped in useEffect, Suspense fallback inside the expanded panel is redundant and will never show.

Use either [Suspense-enabled data loading](https://blog.logrocket.com/react-suspense-data-fetching/) or remove internal Suspense usage to avoid confusion.